skglm/solvers/group_bcd_solver.py

```diff
@@ -95,6 +102,12 @@ def group_bcd_solver(X, y, datafit, penalty, w_init=None, Xw_init=None, p0=10,
             # inplace update of w and Xw
             _bcd_epoch(X, y, w, Xw, datafit, penalty, ws)
 
+            # update intercept
+            if fit_intercept:
+                intercept_old = w[-1]
+                w[-1] -= datafit.intercept_update_step(y, Xw)
+                Xw += (w[-1] - intercept_old)
+
             w_acc, Xw_acc, is_extrapolated = accelerator.extrapolate(w, Xw)
 
             if is_extrapolated:  # avoid computing p_obj for un-extrapolated w, Xw
```

Comment:
probably below, the objective should be computed with w[:n_features]. I wonder why it does not fail, maybe it's not tested